ckanext/canada/templates/public/package/snippets/resource_item.html

```diff
@@ -88,20 +88,14 @@
         </a>
       </li>
       {% endif %}
-      {% if 'fgp_viewer' in pkg.get('display_flags', []) %}
-        {% if res.url and h.is_url(res.url) and res.format not in ['PDF','HTML', 'PNG', 'CSV', 'TXT',] %}
+      {% set fgp_url = h.fgp_viewer_url(pkg) %}
+      {% if fgp_url and res.url and h.is_url(res.url) and res.format not in ['PDF','HTML', 'PNG', 'CSV', 'TXT',] %}
```

Comment:
do you want to pull the format checking logic into the helper too, or does this need to be different in resource_item.html and resources_listh.html?

Reply:
I am displaying the link at two places on the dataset page. One is just relocating the existing "View on Map" button closer to the resource list so that it is more prominent. The other is within the dropdown list for certain resources. I am only doing a format check with the second option.